dbms/src/Common/MPMCQueue.h

```diff
@@ -43,6 +43,9 @@ struct WaitingNode : public SimpleIntrusiveNode<WaitingNode>
 enum class MPMCQueueStatus
 {
     NORMAL,
+    TIMEOUT,
+    EMPTY,
+    FULL,
```

Comment:
How about adding an `MPMCQueueResult` for these new values? Then `MPMCQueueStatus` can keep three values.

Reply:
done